### ISSUE TITLE
Zbarscannerview Android 5.0 freezes

### ISSUE THREAD

**Issue by corest** [dm77/barcodescanner#72]
Hi. I have working scanner application, which uses ZBarScannerView for creating camera preview in activity. My activity created by this example [ScannerActivity.java](https://github.com/dm77/barcodescanner/blob/master/zbar/sample/src/main/java/me/dm7/barcodescanner/zbar/sample/ScannerActivity.java). 
On Android versions 4.X everything works fine and on Android 5.0 sometimes application just freezes. I check catlog and didn't find any errors. In fact - application just stops output any logs after camera preview freezes. After program restart i'm getting black screen on camera preview, because camera resources were not released during this freeze.

**Comment by stephenruda** [comment]
I have been having this issue too. After running the scanner ~1hr my entire app is just frozen and the camera never works until I restart my device.  If anyone knows of a fix please let me know.  I feel like it may be related to this issue:

https://code.google.com/p/android/issues/detail?id=160609&q=camera.open()&colspec=ID%20Type%20Status%20Owner%20Summary%20Stars

**Comment by stephenruda** [comment]
I managed to fix my problems by running all of the computations in onPreviewFrame() on a new background thread and going back to my UI thread with the rawResult when the thread finished

**Comment by Guidgem** [comment]
@stephenruda Can you tell me more about your solution if possible with some code? I will be glad if I can fix this issue because it also bothers me. Thank you! :)

**Comment by stephenruda** [comment]
@Guidgem Unfortunately I thought I had fixed it but it ended up breaking again shortly after I commented  here (and it continue to break).  After plenty of tests on devices that run Android 5.0 or higher, I have determined that it is a bug within Android itself and cannot be fixed by us developers.  I have also ran plenty of tests on devices that run on versions less than 5.0 and they all work just fine.

The freezes always happened very randomly and it never felt like I could do anything about them.

Eventually I just gave up and released my app with the maxSdkVersion set in the manifest; which prevents my app from being seen by 5.0 devices (and higher) within the Play Store.

That is my only recommendation I can give unfortunately.  Hopefully someone comes around and finds a way to fix it but I do really think it is a problem Android. Sorry!

**Comment by Guidgem** [comment]
@stephenruda Thanks for the fast reply. Currently I'm developing in Android 4.4 and I still encounter the same problem. I couldn’t figure out how to fix it either, it seems like it’s a bug in Android as you mentioned. Right know I’m trying to surpass this issue by blocking the refresh of the scannerview when something is scanned with a button to prevent continuous scanning. It’s still a workaround patch and now I’m testing it. Did you change any code from the Zbarscannerview / ZXingcannerview as you mentioned for this version and older?

**Comment by stephenruda** [comment]
@Guidgem Your work around seems like a good idea.  And no, any code I put in to try to fix the issue I eventually pulled out and went back to using an unmodified version of the library.

I had to rollback and flash old versions of Android on my 5.0+ devices to test on them when they are running 4.x.  A few days after I flashed older versions it would still have the same issue it did on v5.0. Strangely enough, they just magically stopped happening (for the most part).  I wouldn't say it is 'perfect' on v4.x but my crash rate was like 5% of what it was when running 5.0.

I'm not really sure what to say about it; it was super frustrating and to this day I still don't know what's going on or if it really is 'fixed'.  I shipped it 'as is' with the maxSdkVersion because for the most part I got the best results on 4.x and rare crashes.
